### PR TITLE
OnRemoveConnection

### PR DESCRIPTION
**Pull request by beddowdm01**
Needed a way to override OnRemoveConnection from a class that inherits from Node like can be done with OnAddConnection. Small change but makes it so much nicer when I need to add functionality when a connection is removed.

### CODE REVIEW

File: Node_Editor_Framework/Runtime/Framework/Core/ConnectionPort.cs [hunk 1]
```diff
@@ -248,6 +248,8 @@ public void RemoveConnection (ConnectionPort port, bool silent = false)
 					() => NodeEditorUndoActions.DeleteConnection(port1, port2),
 					() => NodeEditorUndoActions.CreateConnection(port1, port2),
 					"Delete Connection");
+					
+				port.body.OnRemoveConnection(this, port);
 			}
```
Comment (Seneral) [inline Node_Editor_Framework/Runtime/Framework/Core/ConnectionPort.cs]: Can you move this down to the other callback? Only functional difference is the editor check, but is also more consistent.

**Comment by Seneral** [comment]
Sorry for the late reply. Thanks, no idea why I added OnAddConnection but no remove equivalent. Just one minor change if possible, for consistency and runtime.

File: Node_Editor_Framework/Runtime/Framework/Core/ConnectionPort.cs (hunk 1 above)
Comment (beddowdm01) [reply Node_Editor_Framework/Runtime/Framework/Core/ConnectionPort.cs]: Yeah you're right! Just finished adjusting it, let me know if there's any other changes you'd prefer.

**Comment by Seneral** [comment]
Actually, there are two other things, sorry I didn't see it at first.
First, right now the callback will get the parameters switched, since you're calling it on the connections body.
Second, both port bodies will want the same notification, so we'd need another line:
```
this.body.OnRemoveConnection(this, port);
port.body.OnRemoveConnection(port, this);
```
If in your case the callback should only be received on the one side (e.g. only on output ports), then you can do that by checking the port directions. Other than that there is currently no guaranteed order on which port the connection is removed from.

**Comment by beddowdm01** [comment]
Yeah, makes sense! Apologies for overlooking that :)

**Comment by Seneral** [comment]
No worries, same for me, thanks for the contribution!

**Pull request merged**